src/pybind/mgr/dashboard/frontend/src/app/core/auth/role-form/role-form.component.spec.ts

```diff
@@ -101,6 +101,43 @@ describe('RoleFormComponent', () => {
       roleReq.flush({});
       expect(router.navigate).toHaveBeenCalledWith(['/user-management/roles']);
     });
+
+    it('should check all perms for a scope', () => {
+      form.get('scopes_permissions').setValue({ cephfs: ['read'] });
+      component.onClickCellCheckbox('grafana', 'scope');
+      const scopes_permissions = form.getValue('scopes_permissions');
+      expect(Object.keys(scopes_permissions)).toContain('grafana');
+      expect(scopes_permissions['grafana'].sort()).toEqual(['create', 'delete', 'read', 'update']);
+    });
+
+    it('should uncheck all perms for a scope', () => {
+      form.get('scopes_permissions').setValue({ cephfs: ['read', 'create', 'update', 'delete'] });
+      component.onClickCellCheckbox('cephfs', 'scope');
+      const scopes_permissions = form.getValue('scopes_permissions');
+      expect(Object.keys(scopes_permissions)).not.toContain('cephfs');
+    });
+
+    it('should uncheck all scopes and perms', () => {
+      component.scopes = ['cephfs', 'grafana'];
+      form.get('scopes_permissions').setValue({ cephfs: ['read', 'delete'], grafana: ['update'] });
+      component.onClickHeaderCheckbox('scope', { target: { checked: false } });
+      const scopes_permissions = form.getValue('scopes_permissions');
+      expect(scopes_permissions).toEqual({});
+    });
+
+    it('should check all scopes and perms', () => {
+      component.scopes = ['cephfs', 'grafana'];
+      form
+        .get('scopes_permissions')
+        .setValue({ cephfs: ['create', 'update'], grafana: ['delete'] });
```

Comment:
To set this is not needed as all scopes will be set and don't need to be set before.

Reply:
That is correct, but the intention is to make sure that already selected permissions do not distort the result.